pkg/controller/accountclaim/reuse.go

```diff
@@ -86,9 +86,9 @@ func (r *ReconcileAccountClaim) finalizeAccountClaim(reqLogger logr.Logger, acco
 		return err
 	}
 
-	// Remove IAM user we'll remove the IAM user
+	// Remove IAM user we'll remove the IAM user for CCS
 	if utils.AccountCRHasIAMUserIDLabel(reusedAccount) {
-		err = r.cleanUpIAM(reqLogger, awsClient, reusedAccount, accountClaim)
+		err = account.CleanUpIAM(reqLogger, awsClient, reusedAccount)
 		if err != nil {
 			reqLogger.Error(err, "Failed to delete IAM user during finalizer cleanup")
 		}
```

Comment:
This entire if/else block should be removed, and we should add the account cleanup logic to the account controller finalizer, should be put in this line and the rest of the finalizer moved down: https://github.com/openshift/aws-account-operator/blob/master/pkg/controller/account/account_finalizer.go#L12

Reply:
Chatted with Kirk and clarified that this PR was created to cleanup the IAM code only - the main body of work described in the comment above still remains in [this PR](https://github.com/openshift/aws-account-operator/pull/596) (these 2 PRs were split in an attempt to make it easier on the reviewers).